### DISCUSSION TITLE
Connect customer external domains to vercel subdomain.

### DISCUSSION THREAD

**Discussion by rem4ik4ever** [vercel/community#48] [Help] [answered]
Hello! 

I want to be able to allow my customers to connect their own domains eg. `jimmy.com` and connect with my app so that entering `jommy.com` would show content from my app at `jimmy.myapp.com`. Is there an existing pattern or a guide on how to do so? 

I would much appreciate suggestions and any help thank!

**Comment by wiiiimm** [comment]
Hi,

You will need a [wildcard domain setup](https://vercel.com/docs/custom-domains?query=wildcard#wildcard-domains) and use the [domain api](https://vercel.com/docs/api#endpoints/domains) along with the [alias api](https://vercel.com/docs/api#endpoints/aliases).

**Reply by rem4ik4ever** [reply]
Hey @williamli

So I have setup a project and attached main domain to it `myproject.com` also added wildcard domain `*.myproject.com`. Now I was trying to connect another domain and setup as an alias `anotherdomain.com` => `test.myproject.com`, but it would still point `anotherdomain.com` to `myproject.com`. Can you give me more details on how to properly set this up, please? Thanks!

**Reply by steven-tey** [reply]
Hey @rem4ik4ever! Just answered your question in my comment below with an example that will allow you to do exactly this! Let me know if it helps! :)

**Comment by cill-i-am** [comment]
@rem4ik4ever I have a similar use case. Did you ever figure this out? I'm a bit stuck on how to do it using the answer provided

**Reply by steven-tey** [reply]
@Rocinante89 I just posted an example for this in my comment below, hope it helps!

**Comment by IHIutch** [comment]
Bump, if anyone has a good example of this

**Reply by steven-tey** [reply]
@IHIutch I just posted an example for this in my comment below, hope it helps!

**Reply by IHIutch** [reply]
Thanks! This was mentioned a few days ago in [this thread](https://github.com/vercel/next.js/discussions/12848#discussioncomment-1551776) , and I've already implemented it. Working great!

**Reply by steven-tey** [reply]
Awesome! Glad to hear that you were able to implement it successfully!

**Comment by steven-tey** [comment, marked as answer]
Hey @rem4ik4ever! We recently rolled out a [simple example](https://github.com/vercel/examples/tree/main/edge-functions/hostname-rewrites) that allows you to create unique content pages with a multi-tenant infrastructure using [edge function](https://vercel.com/features/edge-functions) rewrites and [ISR](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration). Each of your users gets assigned a unique subdomain when they create their account, with the (usually paid) option to add a custom domain. 

We're also rolling out more complex boilerplates (e-commerce storefront builders, website builders) + supplementary code snippets (static tweets, domains API, OG image generator) in an [early access program](https://eap.vercel.sh/) – feel free to join if you're interested (click "sign up" on the Platforms on Vercel program).

**Reply by jensmeindertsma** [reply]
Hey @steven-tey 👋. Any chance we can do this when our app is not built with Next.js? 

Something like the following in `vercel.json` maybe?
```json
{
  "rewrites": [
    {
      "source": "/:path*",
      "has": [
        {
          "type": "host",
          "value": ":id.acme.com"
        }
      ],
      "destination": "/users/:id/:path*"
    }
  ]
}
```

**Reply by steven-tey** [reply]
Hey @jensmeindertsma! Don't think you can do rewrites using `vercel.json`...may I know which framework are you using? There should be ways to do it for frameworks like Svelte and Vue – we're looking into that right now! :)

**Reply by rem4ik4ever** [reply]
> Hey @rem4ik4ever! We recently rolled out a [simple example](https://github.com/vercel/examples/tree/main/edge-functions/hostname-rewrites) that allows you to create unique content pages with a multi-tenant infrastructure using [edge function](https://vercel.com/features/edge-functions) rewrites and [ISR](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration). Each of your users gets assigned a unique subdomain when they create their account, with the (usually paid) option to add a custom domain.
> 
> We're also rolling out more complex boilerplates (e-commerce storefront builders, website builders) + supplementary code snippets (static tweets, domains API, OG image generator) in an [early access program](https://eap.vercel.sh/) – feel free to join if you're interested (click "sign up" on the Platforms on Vercel program).

Hey @steven-tey thanks. I managed to solve my problem using server side rendering and checking request domain or subdomain to identify what to show to my customer. I think with edge functions I can improve that functionality!

**Reply by steven-tey** [reply]
@rem4ik4ever awesome! And yeah definitely – with edge functions you can use static generation (ISR/SSG) instead of server-side rendering, which is much faster + leverages caching to reduce the load on your DB/API!
